### ISSUE TITLE
Add :hover treatment to FILE bar when dragging files over it.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1770]
We should give an indication that the file is in the right place and you can drop it to upload it to your project. Currently it shows the browsers's default (+) icon, which shows up no matter where you are hovering in the editor. However, most of the places you drop the file will just end up showing that file in the browser instead of editing it.

**Comment by humphd** [comment]
The code that handles dnd lives here: https://github.com/mozilla/brackets/blob/master/src/utils/DragAndDrop.js

**Comment by pranay414** [comment]
I would like to work on this, can I get a preview on what exactly is needed?

**Comment by humphd** [comment]
@pranay414 it's yours.  @flukeout, do you have some advice on how you want this to look?

**Comment by humphd** [comment]
@pranay414 also note that this will need to happen in the https://github.com/mozilla/brackets repo vs. in Thimble directly, even though it's filed here.  You should start by setting up Brackets on your machine, following the README we have over there.

**Comment by pranay414** [comment]
where is the code where I have to make changes?

**Comment by humphd** [comment]
@pranay414 as I mentioned above, the various drag/drop events are handled in code in https://github.com/mozilla/brackets/blob/master/src/utils/DragAndDrop.js, and you can define a new class in these CSS files: https://github.com/mozilla/brackets/tree/master/src/extensions/default/bramble/stylesheets

**Comment by pranay414** [comment]
<img width="850" alt="screen shot 2017-02-27 at 11 25 46 pm" src="https://cloud.githubusercontent.com/assets/16045021/23373010/1cc81004-fd44-11e6-8e50-4571d4234078.png">
I am getting this error while using 'npm run build', How can i resolve this, is there any other way to work this out?

**Comment by humphd** [comment]
@pranay414 the actual error is above what you pasted.   It looks like it didn't find a file it needs.

**Comment by pranay414** [comment]
I am not able to setup brackets on my machine because of npm error, is there any other way to carry on the development process?

**Comment by humphd** [comment]
@pranay414 if you want help, come talk to us on https://chat.mozillafoundation.org/mozilla/channels/thimble and we can try to solve your install problem.  Without having it setup, you won't be able to do much.

**Comment by shubham0794x** [comment]
I want to work on this issue.

**Comment by pranay414** [comment]
@shubham0794x I am still working on this!

**Comment by humphd** [comment]
@pranay414 did you get your setup issues sorted out?

**Comment by pranay414** [comment]
@humphd Yes everything is fine now and I am working on it, thanks for all your help :)

**Comment by pranay414** [comment]
@humphd I have added the hover treatment to file bar in brackets and same is to be done for thimble so where should I do that?

**Comment by humphd** [comment]
@pranay414 when you same that the same is meant to happen in Thimble, what do you mean?  I think we only need this in the Brackets code.

**Comment by pranay414** [comment]
Oh sorry I got confused, I'll open a PR asap!

**Comment by pranay414** [comment]
"also note that this will need to happen in the https://github.com/mozilla/brackets repo vs. in Thimble directly, even though it's filed here. You should start by setting up Brackets on your machine, following the README we have over there." This thing confused me, just tell me where should i make PR  https://github.com/mozilla/brackets repo or https://github.com/mozilla/thimble.mozilla.org?

**Comment by pranay414** [comment]
@humphd I have made PR in Brackets repo.

**Comment by humphd** [comment]
OK, I did some review in your PR there.  One tip: when you do a PR for an existing issue, reference the issue and what it's about.  For example, with this issue, I'd do this:

```
Add :hover treatment to FILE bar when dragging files over it, fixes https://github.com/mozilla/thimble.mozilla.org/issues/1770
```

When it's an issue in the same repo (usually it is, what we do with Thimble vs. Brackets is not normal), you can do it by giving the issue number like this:

```
Fix #123: Do something to fix some bug, details....
```

**Comment by Pomax** [comment]
Based on the dicussion in https://github.com/mozilla/brackets/pull/638, it was decided to close that PR as the actual nature of the work involved has changed, and requires @humphd and @flukeout to come up with a new strategry we want to follow based on changes we made to how the file tree works/looks/feels.

**Comment by humphd** [comment]
@dnguneratne is going to work on this for us as his first bug, and I'll mentor it.

The way to accomplish this will be to add/remove a class from the filetree component whenever these functions are called (e.g., whenever we handle the drag events):

https://github.com/mozilla/brackets/blob/master/src/utils/DragAndDrop.js#L231-L260

We'll need to figure out what the CSS should look like, especially so that we don't mess up the fact that you can drag files *over* dirs to open them (i.e., we don't want to obscure this by totally covering the drop area).  Maybe we need some kind of outline?  @flukeout can give some ideas.
